oxenstored/connection.ml

```diff
@@ -202,12 +208,24 @@ let get_path con =
   Printf.sprintf "/local/domain/%i/"
     (match con.dom with None -> 0 | Some d -> Domain.get_id d)
 
-let watch_create ~con ~path ~token =
+let get_watch_path_with_base base path =
+  if path.[0] = '@' || path.[0] = '/' then
```

Comment:
I see  lower down a check to see whether a path is relative. Is this used elsewhere? It might make sense to create a couple of functions: `is_absolute` and `is_relative`